Ensure static text fields use the alignment specified by WX on creation.

chris2oph opened the pull request:
The WX type `wxStaticText` was not respecting alignment specified upon creation. This change ensures that we map WX alignment flags to the equivalent QT alignment flags when creating `wxStaticText` types.

Commit:
Ensure static text fields use the alignment specified by WX on creation.

vadz commented on src/qt/stattext.cpp:
```diff
@@ -55,6 +55,34 @@ bool wxStaticText::Create(wxWindow *parent,
+            qt_alignment = Qt::AlignBottom;
+            break;
+        default:
+            qt_alignment = Qt::AlignAbsolute;
```
This is redundant with the initialization of this variable when it's declared, one of them should be removed, ideally.

vadz commented on src/qt/stattext.cpp:
```diff
@@ -55,6 +55,34 @@ bool wxStaticText::Create(wxWindow *parent,
     m_qtLabel->setBuddy( m_qtLabel );
     m_qtLabel->setTextInteractionFlags( Qt::NoTextInteraction );
 
+    Qt::AlignmentFlag qt_alignment = Qt::AlignAbsolute;
```
```suggestion
    Qt::AlignmentFlag qtAlignment = Qt::AlignAbsolute;
```
to conform to the usual naming conventions (alternatively, `alignQt` -- but something in camelCase)

vadz commented on src/qt/stattext.cpp:
```diff
@@ -55,6 +55,34 @@ bool wxStaticText::Create(wxWindow *parent,
+        case wxALIGN_CENTER_HORIZONTAL:
+            qt_alignment = Qt::AlignHCenter;
+            break;
+        case wxALIGN_CENTER_VERTICAL:
```
Is vertical alignment actually supported by this control in Qt? None of the other ports supports it, so I'd just remove this case and the one below. If you do want to support it, you need to do something about `wxALIGN_TOP` too (see the comment for the `switch` above).

vadz commented on src/qt/stattext.cpp:
```diff
@@ -55,6 +55,34 @@ bool wxStaticText::Create(wxWindow *parent,
 
+    Qt::AlignmentFlag qt_alignment = Qt::AlignAbsolute;
+
+    switch (style & wxALIGN_MASK)
```
Using `switch` here is not a good idea because the style can be a combination of bits, e.g., if vertical alignment is supported, it could be `wxALIGN_RIGHT | wxALIGN_BOTTOM`. The other ports just use a sequence of `if`s, in the right order (because `wxALIGN_LEFT` is 0!), see e.g. `wxStaticText::MSWGetStyle()` in `src/msw/stattext.cpp`.

vadz reviewed (changes requested):
Thanks for the changes, but could you please update them as discussed below? TIA!

Commit:
Simplified translation of WX alignment flags to QT alignment flags.

chris2oph commented:
I have addressed your comments @vadz , please let me know if you have any issues with this.

I have removed the SWITCH statement and replaced it with IF-ELSE statements. I've also removed the checks on `wxALIGN_CENTER_VERTICAL` and `wxALIGN_BOTTOM` as per your suggestion.

I have removed the variable for the QT alignment entirely and simply set the QT alignment directly in each separate IF statement.

vadz commented:
Sorry, I still think this could be simplified further by just checking for `wxALIGN_CENTER_HORIZONTAL` once, so I've modified your changes to do it like this, I hope you don't mind.

Thanks!
